web/themes/joinup/joinup_theme.theme

```diff
@@ -202,6 +209,104 @@ function joinup_theme_preprocess_field_multiple_value_form(&$variables) {
 }
 
 /**
+ * Implements hook_preprocess_HOOK() for form-element.html.twig.
+ */
+function joinup_theme_preprocess_form_element(&$variables) {
+  if (isset($variables['label']['#title']) && method_exists($variables['label']['#title'], 'getArguments')) {
+    $arguments = $variables['element']['#title']->getArguments();
+
+    // Array contains form elements, which are displayed with floating label.
+    $floating_label_types = [
+      'email',
+      'entity_autocomplete',
+      'password',
+      'textfield',
+      'url',
+    ];
+
+    if (in_array($variables['type'], $floating_label_types)) {
+      // Make label visible, so it can be displayed inside floating field.
+      $variables['label']['#title_display'] = 'visible';
+
+      // Pass element title to label.
+      if (!empty($arguments['@title'])) {
+        $variables['label']['#title'] = $arguments['@title'];
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter() for input.html.twig.
+ */
+function joinup_theme_theme_suggestions_input_alter(array &$suggestions, array $variables) {
+  // Get input untranslated input value.
+  $input_value = '';
```

Comment:
I don't think it's safe to use labels, even if using the untranslated string is a valid approach. I suggest to switch to checking the element key that was used in the original build array, and use a multi-dimension array for looking that up. Note that this is quite ok for inline entity form things, because they are prefixed by "ief_". Not super sure for the "add_more" one. This code replaces from here to the end of the function.
```php
  // We need to have different styling on some buttons. The key of this
  // multi-dimension array represents the suggestion name, while its values
  // are the build array key of the button being rendered.
  $input_suggestions = [
    'input__ief_add_another' => [
      // Generic "Add more" button for multi-value widgets.
      'add_more',
      'ief_add_existing',
      'ief_add',
      'ief_reference_save',
      'ief_add_save',
      'ief_edit_save',
    ],
    'input__ief_edit' => [
      'ief_entity_edit',
      'ief_entity_edit_save',
    ],
    'input__ief_remove' => [
      'ief_entity_remove',
      'ief_remove_confirm',
    ],
    'input__ief_cancel' => [
      'ief_entity_cancel',
      'ief_remove_cancel',
    ],
    'input__ief_cancel_circle' => [
      'ief_add_cancel',
      'ief_edit_cancel',
      'ief_reference_cancel',
    ],
  ];

  // The last key of #array_parents is the build array key used in the code.
  $element_key = end($variables['element']['#array_parents']);
  foreach ($input_suggestions as $suggestion => $names) {
    if (in_array($element_key, $names)) {
      $suggestions[] = $suggestion;
    }
  }
```